src/time.rs

```diff
@@ -395,14 +402,18 @@ impl TryConvert for SystemTime {
             timespec = rb_time_timespec(val.as_rb_value());
             Ruby::get_unchecked().qnil()
         })?;
-        if timespec.tv_sec >= 0 && timespec.tv_nsec >= 0 {
-            let mut duration = Duration::from_secs(timespec.tv_sec as _);
+        if timespec.tv_nsec >= 0 {
+            let mut duration = Duration::from_secs(timespec.tv_sec.abs() as _);
             duration += Duration::from_nanos(timespec.tv_nsec as _);
-            Ok(Self::UNIX_EPOCH + duration)
+            if timespec.tv_sec >= 0 {
+                Ok(Self::UNIX_EPOCH + duration)
+            } else {
+                Ok(Self::UNIX_EPOCH - duration)
+            }
         } else {
             Err(Error::new(
                 Ruby::get_with(val).exception_arg_error(),
-                "time must not be negative",
+                "time nanos must not be negative",
             ))
```

Comment:
While we're here, I took the liberty of adding support for converting Ruby `Time` objects for dates prior to 1970 to `SystemTime`.  I believe Ruby's [normalization of timespecs](https://github.com/ruby/ruby/blob/e798f45cc2417ae778e4790e3159654c90ba91fe/time.c#L2693-L2700) will make it unlikely that we'll get a `timespec` with negative nanoseconds, but I'm not confident of that.

Reply:
Now that you've added support for taking a timespec and providing an offset, a question that occurs to me is whether a `Time` object created from a `SystemTime` should start out in UTC instead of the system timezone.

Reply:
I agree that it seems extremely unlikely that we'll get a `timespec` with negative nanos, but in this case I don't think there's any harm in having the error handling there just in case.

The way I think about it, Time is a value (seconds from epoch), and a label on how to interpret it (the zone). Like how a string is a value (some bytes) and a label (the encoding).

With my way of thinking about it SystemTime is just a value, without the label (like `[u8]` vs `String`). Ruby's Time doesn't have the option to omit the label, so what's the default label?
I'd probably go with UTC being the sensible default (but there might be some bias there having grown up in England basically on the prime meridian). However I think there's a pretty strong case for Ruby having picked local time as the default, so I think we should stick with local.